src/gui/maptools/qgsmaptoolcapture.cpp

```diff
@@ -492,11 +492,15 @@ void QgsMapToolCapture::setCurrentShapeMapTool( const QgsMapToolShapeMetadata *s
 
 void QgsMapToolCapture::cadCanvasMoveEvent( QgsMapMouseEvent *e )
 {
+  // If we are adding a record to a non-spatial layer, just return
+  if ( mCaptureModeFromLayer && !canvas()->currentLayer()->isSpatial() )
```

Comment:
@nicogodet 
so only a matter of adding:
`  if ( mCaptureModeFromLayer && ( !canvas()->currentLayer() || !canvas()->currentLayer()->isSpatial() ) )`

Reply:
@3nids Can't confirm, I don't have a build setup right know :/